### PR TITLE
ci: All jobs should pass for CI to pass

### PR DESCRIPTION
**Pull request by mrobinson**
This change makes it so that CI does not pass unless all jobs defined in
the workflow file pass. This is necessary in order to prevent landing
changes that break CI jobs. `success()` and `failure()` which were used
before do not take into account dependent jobs.

### CODE REVIEW

**Review by rillian — APPROVED** [review]
So the issue is that the merge queue accepted #506 even though the lint job failed? If so this should fix it.

If I understand correctly, the `build_results` job is a way to encode which jobs are required in-tree, so only that job needs to be marked as required in the github settings. It's an extra review burden to check this is set correctly, but as far as I can tell, there's no way to require *all* status checks in a branch protection rule?

**Comment by mrobinson** [comment]
@rillian Thanks for the review. This will also need a review from someone with write access to the repository. I think that for branch protection you have to manually add all jobs that you'd like to gate landing a change. We typically just use this "meta job" approach in Servo. Unfortunately, some of the "meta jobs" are not written correctly, as is the case here.

**Review by mukilan — APPROVED** [review]

**Pull request merged**
